AEPIdentity/Sources/Identity.swift

```diff
@@ -55,41 +55,50 @@ import Foundation
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
-        guard canProcessEvents(event: event) else { return false }
+        guard let state = state else { return false }
+
+        // fast boot identity without waiting for configuration
+        if state.boot(event: event, createSharedState: createSharedState(data:event:)) {
+
+            // skip waiting for configuration if event is getExperienceCloudId event or getIdentifiers event
+            if event.isGetIdentifierEvent {
+                Log.trace(label: "\(name):\(#function)", "Not waiting for latest configuration as we have the ECID cached and available for processing [event:(\(event.name)) id:(\(event.id)].")
+                return true
+            }
+        }
+
+        guard state.forceSyncIdentifiers(configSharedState: getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value, event: event, createSharedState: createSharedState(data:event:)) else { return false }
```

Comment:
As we discussed since the extension does not have access to the events queue, we should not do anything to change the order of the events here.
The only change should be when a getIdentifierEvent is received we don't check for config. The sequence would look like:
```
state.boot(event: event, createSharedState: createSharedState(data:event:))
if !state.forceSyncIdentifiers(configSharedState: getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value, event: event, createSharedState: createSharedState(data:event:)) else { return false }
if event.isGetIdentifierEvent { return true } // does not wait for config
if event.isSyncEvent ...
```

This should also account for the case where ECID is not generated yet on first launch, which will be generated on the forceSyncIdentifiers step.

Reply:
forceSyncIdentifiers requires valid config and if we have pending shared state then this will again not work right? Is the change for not setting pending shared state for configuration going to happen?

Reply:
@addb we should keep the two changes separately: this one is specifically to return readyForEvent without waiting for config and the not setting pending shared state changes will solve for having the config resolved quickly, so even if the getIdentifier event is queued after a force sync event it will return pretty quickly.

Reply:
@emdobrin Even without setting a pending state, when the SDK starts, if an app has no bundled config and only configures with app id, then readyForEvent will still return false until that first configuration request is resolved. From your recommendation, the shared state is created with the persisted ECID immediately, but any getIdentifier events will wait for the initial configuration. I'm not clear on that separation and why we don't handle both at the same time, either both after `state.boot` or after the force sync event.

Reply:
Sharing the initial shared state on boot is still required for other extensions or data elements to consume if it already has identities, so that change by itself will improve certain workflows.
With regards to not retuning on getters right away on boot, I do not see a reason why we should speed up the process for get identities events only if they are received first by this extension (on boot), but not in other use-cases, that could become difficult to debug and maintain over time. 
However, I 💯  agree we should revisit this point if consuming the cached configuration and not block on pending state does not solve for this use-case.